Khan/wonder-blocks
# FEI-4022: implement useTimeout hook

## Summary:
This hook follows similar patterns to ActionScheduler.  I had to tweak things a little be since we're dropping the need for the scheduler itself with this hook.

I've also added lint rules for hooks in this PR.  I'll create a separate PR with that change and rebase this one on top of it.

Issue: FEI-4022

## Test plan:
- yarn jest use-timeout.test.js
- yarn start:storybook
- load http://localhost:6061/?path=/docs/timing-usetimeout
- see that the timeout starts immediately with the first example
- see that the timeout doesn't start until you click 'set' in the seconde example
- play around with setting and clearing the timeouts
